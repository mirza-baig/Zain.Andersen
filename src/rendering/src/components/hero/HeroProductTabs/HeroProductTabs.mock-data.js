const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'HeroProductTabs',
  },
  fields: {
    headline: {
      value: 'Hero Product Tabs Headline',
    },
    sticky: {
      value: false,
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
    children: [
      {
        id: '1f4ee7ff-e892-491c-8a5e-17c367f6460c',
        displayName: 'Hero Product Tab Item 1',
        name: 'Hero Product Tab Item 1',
        templateId: '8b70967e-a2ac-4f92-a099-2584e1d30523',
        templateName: 'Hero Product Tab Item',
        url: '/Components/Hero-Product-Tabs/Data/Hero-Product-Tabs/Hero-Product-Tab-Item-1',
        fields: {
          title: {
            value: 'Hero Product Tab Item 1',
          },
          contentId: {
            value: '',
          },
          children: [],
        },
        placeholders: {
          components: [
            {
              componentName: 'PromoGeneric',
              uid: '8d0b5ef9-af0c-4874-9f8a-26483046665b',
              dataSource: '{8D0B5EF9-AF0C-4874-9F8A-26483046665B}',
              fields: {
                bottomCaptionDescription: {
                  value:
                    '\n<span id="fld_6649519182FE4419BC5C18EA73D49284_0C243F9602584CEFB0027280B78FB028_en_1_c13e08852c664e6fbd123704a2b98820_196365_edit" sc_parameters="prevent-line-break=true" class="scWebEditInput scEnabledChrome" scfieldtype="single-line text" scdefaulttext="[No text in field]" sc-part-of="field" scwatermark="true" contenteditable="true">Bottom description goes here and spans to two line text over here dummy text goes here</span>\n\n',
                },
                bottomCaptionHeadline: {
                  value:
                    'Bottom top copy goes here and spans to two line text over here dummy text goes here',
                },
                imageRatio: null,
                componentSpacing: null,
                sectionId: {
                  value: '',
                },
                imgPosition: {
                  id: '0a58d643-e0c1-4a6e-a9f6-03d731a55121',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Right',
                  name: 'Right',
                  displayName: 'Right',
                  fields: {
                    Value: {
                      value: 'right',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                headlineLevel: {
                  id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
                  name: 'Heading 2',
                  displayName: 'Heading 2',
                  fields: {
                    Value: {
                      value: 'h2',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                headlineText: {
                  value: 'Headline text goes here  with right image 1:1 ',
                },
                eyebrowLevel: null,
                eyebrowText: {
                  value: '',
                },
                body: {
                  value:
                    "<p><strong>Lorem Ipsum</strong> is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.</p>",
                },
                cta1Icon: {
                  id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
                  name: 'Arrow',
                  displayName: 'Arrow',
                  fields: {
                    Value: {
                      value: 'arrow',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                cta1Link: {
                  value: {
                    href: '/Demo-HeroTwoColumn',
                    linktype: 'internal',
                    text: 'Call to Action',
                    querystring: '',
                    target: '',
                    id: '{0AF8C545-074B-466E-93CD-B14BF83DB353}',
                  },
                },
                cta1Modal: null,
                cta1ModalLinkText: {
                  value: '',
                },
                cta1Style: {
                  id: '49a23327-0397-4cce-a930-e76918d37c42',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
                  name: 'Primary',
                  displayName: 'Primary',
                  fields: {
                    Value: {
                      value: 'primary',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                primaryImage: {
                  value: {
                    src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Picture/photo-1581141849291-1125c7b692b5.jpg?h=900&iar=0&w=1600',
                    alt: 'Img',
                    width: '1600',
                    height: '900',
                  },
                },
                primaryImageMobile: {
                  value: {},
                },
                primaryImageMobileFocusArea: null,
                primaryVideo: null,
                eventName: {
                  value: '',
                },
                eventType: {
                  value: '',
                },
                eventZone: {
                  value: '',
                },
              },
            },
          ],
        },
      },
      {
        id: 'bab12330-424f-4856-a16c-d87011fa5b20',
        displayName: 'Hero Product Tab Item 2',
        name: 'Hero Product Tab Item 2',
        templateId: '8b70967e-a2ac-4f92-a099-2584e1d30523',
        templateName: 'Hero Product Tab Item',
        url: '/Components/Hero-Product-Tabs/Data/Hero-Product-Tabs/Hero-Product-Tab-Item-2',
        fields: {
          title: {
            value: 'Hero Product Tab Item 2',
          },
          contentId: {
            value: '',
          },
          children: [],
        },
        placeholders: {
          components: [
            {
              componentName: 'PromoGeneric',
              uid: 'c93358fe-77f9-4af8-9e10-eeecaf8cdae8',
              dataSource: '{C93358FE-77F9-4AF8-9E10-EEECAF8CDAE8}',
              fields: {
                bottomCaptionDescription: {
                  value:
                    '<p>Bottom description goes here and spans to two line text over here dummy text goes here</p>',
                },
                bottomCaptionHeadline: {
                  value:
                    'Bottom top copy goes here and spans to two line text over here dummy text goes here',
                },
                imageRatio: null,
                componentSpacing: null,
                sectionId: {
                  value: '',
                },
                imgPosition: {
                  id: '0a58d643-e0c1-4a6e-a9f6-03d731a55121',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Right',
                  name: 'Right',
                  displayName: 'Right',
                  fields: {
                    Value: {
                      value: 'right',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                headlineLevel: {
                  id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
                  name: 'Heading 2',
                  displayName: 'Heading 2',
                  fields: {
                    Value: {
                      value: 'h2',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                headlineText: {
                  value: 'Headline goes here with right image 3:2 ',
                },
                eyebrowLevel: null,
                eyebrowText: {
                  value: '',
                },
                body: {
                  value:
                    '<strong>Lorem Ipsum</strong> is simply dummy text of the printing and\ntypesetting industry.&nbsp;',
                },
                cta1Icon: {
                  id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
                  name: 'Arrow',
                  displayName: 'Arrow',
                  fields: {
                    Value: {
                      value: 'arrow',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                cta1Link: {
                  value: {
                    href: '/Demo-HeroTwoColumn',
                    linktype: 'internal',
                    text: 'Call to Action',
                    querystring: '',
                    target: '',
                    id: '{0AF8C545-074B-466E-93CD-B14BF83DB353}',
                  },
                },
                cta1Modal: null,
                cta1ModalLinkText: {
                  value: '',
                },
                cta1Style: {
                  id: '49a23327-0397-4cce-a930-e76918d37c42',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
                  name: 'Primary',
                  displayName: 'Primary',
                  fields: {
                    Value: {
                      value: 'primary',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                primaryImage: {
                  value: {
                    src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Picture/photo-1581141849291-1125c7b692b5.jpg?h=900&iar=0&w=1600',
                    alt: 'Img',
                    width: '1600',
                    height: '900',
                  },
                },
                primaryImageMobile: {
                  value: {},
                },
                primaryImageMobileFocusArea: null,
                primaryVideo: null,
                eventName: {
                  value: '',
                },
                eventType: {
                  value: '',
                },
                eventZone: {
                  value: '',
                },
              },
            },
          ],
        },
      },
      {
        id: 'bb1afcdd-d743-45a8-a2f3-c3426f91cdf9',
        displayName: 'Hero Product Tab Item 3',
        name: 'Hero Product Tab Item 3',
        templateId: '8b70967e-a2ac-4f92-a099-2584e1d30523',
        templateName: 'Hero Product Tab Item',
        url: '/Components/Hero-Product-Tabs/Data/Hero-Product-Tabs/Hero-Product-Tab-Item-3',
        fields: {
          title: {
            value: 'Hero Product Tab Item 3',
          },
          contentId: {
            value: '',
          },
          children: [],
        },
        placeholders: {
          components: [
            {
              componentName: 'PromoGeneric',
              uid: '3ca91ad4-261a-4745-aaf4-e7ff740aeb24',
              dataSource: '{3CA91AD4-261A-4745-AAF4-E7FF740AEB24}',
              fields: {
                bottomCaptionDescription: {
                  value: '',
                },
                bottomCaptionHeadline: {
                  value: '',
                },
                imageRatio: null,
                componentSpacing: null,
                sectionId: {
                  value: '',
                },
                imgPosition: null,
                headlineLevel: {
                  id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
                  url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
                  name: 'Heading 2',
                  displayName: 'Heading 2',
                  fields: {
                    Value: {
                      value: 'h2',
                    },
                  },
                  templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                  templateName: 'Enum',
                },
                headlineText: {
                  value: '',
                },
                eyebrowLevel: null,
                eyebrowText: {
                  value: '',
                },
                body: {
                  value: '',
                },
                cta1Icon: null,
                cta1Link: {
                  value: {
                    href: '',
                  },
                },
                cta1Modal: null,
                cta1ModalLinkText: {
                  value: '',
                },
                cta1Style: null,
                primaryImage: {
                  value: {
                    src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Picture/photo-1581141849291-1125c7b692b5.jpg?h=900&iar=0&w=1600',
                    alt: 'Img',
                    width: '1600',
                    height: '900',
                  },
                },
                primaryImageMobile: {
                  value: {},
                },
                primaryImageMobileFocusArea: null,
                primaryVideo: null,
                eventName: {
                  value: '',
                },
                eventType: {
                  value: '',
                },
                eventZone: {
                  value: '',
                },
              },
            },
          ],
        },
      },
    ],
  },
};

export default defaultData;
