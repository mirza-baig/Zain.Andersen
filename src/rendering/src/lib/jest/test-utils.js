// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
