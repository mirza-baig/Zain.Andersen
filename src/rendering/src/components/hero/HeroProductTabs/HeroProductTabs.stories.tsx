// Global
import { Story, Meta } from '@storybook/react';
// Local
import HeroProductTabs, { HeroProductTabsProps } from './HeroProductTabs';
import defaultData from './HeroProductTabs.mock-data';
import fourTabsData from './HeroProductTabs.mock-data-with-four-tabs';

export default {
  title: 'Hero/HeroProductTabs',
  component: HeroProductTabs,
} as Meta;
const Template: Story<HeroProductTabsProps> = (props) => <HeroProductTabs {...props} />;

export const Default = Template.bind({});
Default.args = defaultData;

export const fourTabs = Template.bind({});
fourTabs.args = fourTabsData;
