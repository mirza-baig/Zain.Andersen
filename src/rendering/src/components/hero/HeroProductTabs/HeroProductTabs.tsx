// Global
import { useEffect, useState, useRef } from 'react';
import { useRouter } from 'next/router';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import {
  Placeholder,
  withDatasourceCheck,
  Text,
  RichText,
} from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { HeroProductTabsTheme } from './HeroProductTabs.theme';
import { Tabs, TabList, TabPanel, resetIdCounter } from 'react-tabs';
import classNames from 'classnames';
import { SliderWrapper } from 'src/helpers/SliderWrapper';
import { useCurrentScreenType } from 'lib/utils/get-screen-type';

export type HeroProductTabsProps =
  Feature.EnterpriseWeb.Enterprise.Components.Hero.HeroProductTabs.HeroProductTabs;
const HeroProductTabs = (props: HeroProductTabsProps) => {
  const { fields } = props;
  resetIdCounter();
  const slidesToShow = 2;
  const initialSlide = 0;
  const numberOfTabs = props.fields?.children.length;

  const componentRef = useRef<HTMLDivElement>(null);
  const getHashAnchor = (): number => {
    let hashedTabIndex = -1;
    if (typeof window !== 'undefined' && window.location.hash !== '') {
      const hash = window.location.hash.replace('#', '').toLowerCase();
      hashedTabIndex = props.fields?.children.findIndex((child: HeroProductTabsProps) => {
        const childId = child.id.toLowerCase();
        const contentId = child.fields?.contentId?.value?.toLowerCase();
        return childId === hash || contentId === hash;
      });
    }
    return hashedTabIndex;
  };

  let defaultTab = 0;

  useEffect(() => {
    const handleTabChange = () => {
      const hash = window.location.hash.replace('#', '');
      const hashedTab = getHashAnchor();
      if (hashedTab > -1) {
        setTabIndex(hashedTab);
      } else if (!!props.fields?.defaultActiveTab) {
        const defaultTabIndex = props.fields?.children.findIndex((child: HeroProductTabsProps) => {
          return hash === child.id;
        });
        setTabIndex(defaultTabIndex > 0 ? defaultTabIndex : defaultTab);
      }
    };

    const handleWindowResize = () => {
      // Set specific height of slick-track so tabs can fill height based on other tabs content
      document.querySelectorAll<HTMLElement>('.heroproducttabs .slick-track').forEach((track) => {
        track.style.height = track.offsetHeight.toString() + 'px';
      });
    };

    const handleScrollToTab = () => {
      if (getHashAnchor() > -1 && componentRef.current) {
        componentRef.current.scrollIntoView();
      }
    };

    const hash = router && router.asPath.split('#')[1];
    let defaultTab = 0;
    const hashedTab = getHashAnchor();
    if (hashedTab > -1) {
      defaultTab = hashedTab;
    } else if (!!props.fields?.defaultActiveTab) {
      const defaultTabIndex = props.fields?.children.findIndex(() => {
        return hash === props.fields?.defaultActiveTab?.id;
      });
      defaultTab = defaultTabIndex > 0 ? defaultTabIndex : defaultTab;
    }

    setTabIndex(defaultTab);
    handleTabChange();

    window.addEventListener('hashchange', handleTabChange);
    window.addEventListener('resize', handleWindowResize);

    handleScrollToTab();

    return () => {
      window.removeEventListener('hashchange', handleTabChange);
      window.removeEventListener('resize', handleWindowResize);
    };
    // "props.fields?.children" and "props.fields?.defaultActiveTab" are coming from layout.
    // "getHashAnchor" is a function which is not going to change.
    // "router" is not required as dependency as we only need this code block to be executed only on first render.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const [tabIndex, setTabIndex] = useState(0);

  const router = useRouter();

  const { screenType } = useCurrentScreenType();

  const handleAfterChange = (index: number) => {
    const fadeDiv0 = document.querySelector('.fade-div-0');
    const fadeDiv3 = document.querySelector('.fade-div-3');
    if (fadeDiv0 && fadeDiv3 && index === 1 && (screenType === 'md' || screenType === 'sm')) {
      fadeDiv0.classList.remove('hidden');
      fadeDiv0.classList.add('block');
      fadeDiv3.classList.add('hidden');
    }
    if (fadeDiv3 && fadeDiv0 && index == 0 && (screenType === 'md' || screenType === 'sm')) {
      fadeDiv3.classList.remove('hidden');
      fadeDiv3.classList.add('block');
      fadeDiv0.classList.add('hidden');
    }
  };

  const sliderSettings = {
    infinite: false,
    speed: 500,
    initialSlide: initialSlide,
    slidesToShow: numberOfTabs > slidesToShow ? slidesToShow : 2,
    slidesToScroll: 1,
    dots: false,
    arrows: false,
    variableWidth: true,
    className: 'heroproducttabs',
    swipeToSlide: true,
    afterChange: handleAfterChange,
  };

  const TabHeading = ({ onClick, children, index }: HeroProductTabsProps) => {
    return (
      <div
        className={classNames(
          themeData.classes.tabs.commonTabHeadingClass,
          index === tabIndex
            ? themeData.classes.tabs.selectedTabHeadingClass
            : themeData.classes.tabs.unselectedTabHeadingClass
        )}
        onClick={onClick}
        role="tab"
      >
        {children}
      </div>
    );
  };
  TabHeading.tabsRole = 'Tab';

  const { themeName, themeData } = useTheme(HeroProductTabsTheme(screenType ?? 'sm', numberOfTabs));

  const isStickyTabonPage = fields?.sticky.value;

  useEffect(() => {
    const handleCTAClick = (event: MouseEvent) => {
      const target = event.target as Element;

      if (target && target.tagName === 'A' && target.getAttribute('href')?.includes('#')) {
        // determine if path given resides on the page
        const currPathname = window.location.pathname;
        const targetPathname = target.getAttribute('href')?.split('#')[0];

        if (currPathname === targetPathname || targetPathname === '') {
          event.preventDefault();

          const currentURL = window.location.href;
          const hash = target.getAttribute('href')?.split('#')[1];
          const newURL = currentURL.split('#')[0] + '#' + hash;

          window.location.assign(newURL);

          // eslint-disable-next-line react-hooks/exhaustive-deps
          defaultTab = 0;
          const hashedTab = getHashAnchor();
          if (hashedTab > -1) {
            defaultTab = hashedTab;
          } else if (!!props.fields?.defaultActiveTab) {
            const defaultTabIndex = props.fields?.children.findIndex(() => {
              return hash === props.fields?.defaultActiveTab?.id;
            });
            defaultTab = defaultTabIndex > 0 ? defaultTabIndex : defaultTab;
          }
          setTabIndex(defaultTab);
        }
      }
    };

    if (router) {
      const getRAQurl = router.asPath;
      const fragmentIndex = getRAQurl.indexOf('#'); // Get the index of the "#" character
      const fragment = fragmentIndex !== -1 ? getRAQurl.substring(fragmentIndex) : ''; // Extract the fragment if present

      if (fragment == '#request_a_quote') {
        setTabIndex(1);
      }
    }
    document.addEventListener('click', handleCTAClick);

    return () => {
      document.removeEventListener('click', handleCTAClick);
    };
  }, []);

  const stickyTabClass = (theme: string) => {
    if (!isStickyTabonPage) {
      return '';
    }

    return `sticky ${theme === 'aw' ? 'top-[55px]' : 'top-0'} ml:top-[122px]`;
  };

  if (!fields) {
    return null;
  }
  return (
    <Component padding={'0'} variant="full" dataComponent="hero/heroproducttabs" {...props}>
      <div className=" col-span-12" ref={componentRef}>
        <input type="hidden" value="brianc"></input>
        <Tabs selectedIndex={tabIndex} onSelect={(index: number) => setTabIndex(index)}>
          {/* Added z-index from 70 to 19 it was conflicts with search popup modal on desktop view */}
          <div
            className={`heroproducttabs-wrapper z-[19] w-full border-b border-b-gray bg-white ${stickyTabClass(
              themeName
            )}`}
            role="tablist"
          >
            <div className="flex w-full flex-col items-center justify-between ml:flex-row ml:px-m lg:mx-auto lg:max-w-screen-lg">
              <div className="mr-s w-full ml:w-fit">
                <RichText
                  tag={'h1'}
                  className={themeData.classes.productHeadlineClass}
                  field={fields.headline}
                />
              </div>
              <div className="ml-auto w-full max-ml:border-t max-ml:border-t-gray ml:w-fit">
                <TabList>
                  {screenType === 'sm' ? (
                    <SliderWrapper sliderSettings={sliderSettings}>
                      {props.fields.children.map((tab: HeroProductTabsProps, index: number) => {
                        const tabId = `react-tabs-${index}`;
                        const ariaLabelledby = `${tabId}-label`;
                        return (
                          <TabHeading
                            key={index}
                            index={index}
                            onClick={() => {
                              setTabIndex(index);
                              window.location.hash =
                                tab.fields?.contentId?.value !== ''
                                  ? tab.fields?.contentId?.value
                                  : tab.id;
                            }}
                            aria-labelledby={ariaLabelledby}
                            role="tab"
                          >
                            <Text
                              tag={'h3'}
                              className={classNames(
                                themeData.classes.tabs.headlineClass,
                                index === tabIndex
                                  ? themeData.classes.tabs.selectedHeadlineClass
                                  : ''
                              )}
                              field={tab.fields.title}
                            />
                            {numberOfTabs > 3 && (index === 0 || index === 3) && (
                              <div
                                className={classNames(
                                  'absolute left-0 top-0 h-full w-full content-[""]',
                                  index === 3 && tabIndex === 0 ? 'block' : 'hidden',
                                  'fade-div-' + index
                                )}
                                style={{
                                  background:
                                    'linear-gradient(270deg, #ffffff 0%, rgba(255, 255, 255, 0) 100%)',
                                }}
                              ></div>
                            )}
                          </TabHeading>
                        );
                      })}
                    </SliderWrapper>
                  ) : (
                    <div className="flex flex-row">
                      {props.fields.children.map((tab: HeroProductTabsProps, index: number) => {
                        const tabId = `react-tabs-${index}`;
                        const ariaLabelledby = `${tabId}-label`;
                        return (
                          <TabHeading
                            key={index}
                            index={index}
                            onClick={() => {
                              setTabIndex(index);
                              window.location.hash =
                                tab.fields?.contentId?.value !== ''
                                  ? tab.fields?.contentId?.value
                                  : tab.id;
                            }}
                            aria-labelledby={ariaLabelledby}
                            role="tab"
                          >
                            <Text
                              tag={'h3'}
                              className={classNames(
                                themeData.classes.tabs.headlineClass,
                                index === tabIndex
                                  ? themeData.classes.tabs.selectedHeadlineClass
                                  : ''
                              )}
                              field={tab.fields.title}
                            />
                          </TabHeading>
                        );
                      })}
                    </div>
                  )}
                </TabList>
              </div>
            </div>
          </div>
          {props.fields.children.map((tab: HeroProductTabsProps, index: number) => {
            const tabId = `react-tabs-${index}`;
            const ariaLabelledby = `${tabId}-label`;
            return (
              <TabPanel forceRender={true} key={index} id={tabId} aria-labelledby={ariaLabelledby}>
                <Placeholder
                  name="components"
                  rendering={tab}
                  render={(tabs) =>
                    tabs.map((tab, index) => (
                      <div key={index} className={classNames('col-span-12')}>
                        {tab}
                      </div>
                    ))
                  }
                  favoriteProducts={props.favoriteProducts}
                />
              </TabPanel>
            );
          })}
        </Tabs>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<HeroProductTabsProps>(HeroProductTabs);
