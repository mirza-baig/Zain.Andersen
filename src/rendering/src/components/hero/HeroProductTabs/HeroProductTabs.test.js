// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import HeroProductTabs from './HeroProductTabs';
import defaultData from './HeroProductTabs.mock-data';

it('renders correctly', async () => {
  const component = snapshot(HeroProductTabs, { componentProps: defaultData });
  await hasDataComponent(component, 'hero/heroproducttabs');
});
