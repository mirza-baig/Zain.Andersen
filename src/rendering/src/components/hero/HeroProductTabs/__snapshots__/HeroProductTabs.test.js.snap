// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="hero/heroproducttabs"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 0 gap-s md:gap-s "
    >
      <div
        class=" col-span-12"
      >
        <input
          type="hidden"
          value="brianc"
        />
        <div
          class="react-tabs"
          data-rttabs="true"
        >
          <div
            class="heroproducttabs-wrapper z-[19] w-full border-b border-b-gray bg-white "
            role="tablist"
          >
            <div
              class="flex w-full flex-col items-center justify-between ml:flex-row ml:px-m lg:mx-auto lg:max-w-screen-lg"
            >
              <div
                class="mr-s w-full ml:w-fit"
              >
                <h1
                  class="text-sm-s lg:text-s font-heavy font-sans uppercase max-ml:pl-ml max-ml:pb-xxs"
                >
                  Hero Product Tabs Headline
                </h1>
              </div>
              <div
                class="ml-auto w-full max-ml:border-t max-ml:border-t-gray ml:w-fit"
              >
                <ul
                  class="react-tabs__tab-list"
                  role="tablist"
                >
                  <div
                    class="flex flex-row"
                  >
                    <div
                      class="react-tabs__tab flex h-full ml:w-fit lg:max-w-[250px]  mr-s last-of-type:mr-0  relative justify-center text-center items-center px-xxs py-xs md:px-s border-b-8 w-auto react-tabs__tab--selected border-b-primary "
                      role="tab"
                    >
                      <h3
                        class="font-heavy font-sans uppercase text-sm-xs ml:text-sm-s text-dark-gray hover:text-black text-base lg:text-sm-s justify-center text-center text-black"
                      >
                        Hero Product Tab Item 1
                      </h3>
                    </div>
                    <div
                      class="react-tabs__tab flex h-full ml:w-fit lg:max-w-[250px]  mr-s last-of-type:mr-0  relative justify-center text-center items-center px-xxs py-xs md:px-s border-b-8 w-auto text-dark-gray border-b-white  "
                      role="tab"
                    >
                      <h3
                        class="font-heavy font-sans uppercase text-sm-xs ml:text-sm-s text-dark-gray hover:text-black text-base lg:text-sm-s justify-center text-center"
                      >
                        Hero Product Tab Item 2
                      </h3>
                    </div>
                    <div
                      class="react-tabs__tab flex h-full ml:w-fit lg:max-w-[250px]  mr-s last-of-type:mr-0  relative justify-center text-center items-center px-xxs py-xs md:px-s border-b-8 w-auto text-dark-gray border-b-white  "
                      role="tab"
                    >
                      <h3
                        class="font-heavy font-sans uppercase text-sm-xs ml:text-sm-s text-dark-gray hover:text-black text-base lg:text-sm-s justify-center text-center"
                      >
                        Hero Product Tab Item 3
                      </h3>
                    </div>
                  </div>
                </ul>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-tabs-0"
            class="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-1"
            role="tabpanel"
          >
            <div
              class="col-span-12"
            >
              <div
                style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
              >
                <h2>
                  PromoGeneric
                </h2>
                <p>
                  JSS component is missing React implementation. See the developer console for more information.
                </p>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-tabs-2"
            class="react-tabs__tab-panel"
            id="react-tabs-3"
            role="tabpanel"
          >
            <div
              class="col-span-12"
            >
              <div
                style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
              >
                <h2>
                  PromoGeneric
                </h2>
                <p>
                  JSS component is missing React implementation. See the developer console for more information.
                </p>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-tabs-4"
            class="react-tabs__tab-panel"
            id="react-tabs-5"
            role="tabpanel"
          >
            <div
              class="col-span-12"
            >
              <div
                style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
              >
                <h2>
                  PromoGeneric
                </h2>
                <p>
                  JSS component is missing React implementation. See the developer console for more information.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
